compiler/rustc_typeck/src/collect.rs

```diff
@@ -2724,6 +2726,15 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
             codegen_fn_attrs.flags |= CodegenFnAttrFlags::NAKED;
         } else if tcx.sess.check_name(attr, sym::no_mangle) {
             codegen_fn_attrs.flags |= CodegenFnAttrFlags::NO_MANGLE;
+        } else if attr.has_name(sym::feature) {
+            if let Some(list) = attr.meta_item_list() {
+                if list.iter().any(|nested_meta_item| nested_meta_item.has_name(sym::no_coverage)) {
+                    tcx.sess.mark_attr_used(attr);
+                    no_coverage_feature_enabled = true;
+                }
```

Comment:
Shouldn't AssumeUsed make this unnecessary?

Reply:
No, it appears to be necessary. I tried removing it, and when compiling rustc I now get the error:

```

error: unused attribute
   --> library/core/src/cmp.rs:277:32
    |
277 |     #[cfg_attr(not(bootstrap), feature(no_coverage))]
    |                                ^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D unused-attributes` implied by `-D warnings`
```